src/libraries/System.IO.Compression/tests/ZipArchive/zip_LargeFiles.cs

```diff
@@ -0,0 +1,48 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using System.Linq;
+using Xunit;
+
+namespace System.IO.Compression.Tests
+{
+    [Collection(nameof(DisableParallelization))]
+    public class zip_LargeFiles : ZipFileTestBase
+    {
+        [ConditionalFact(typeof(PlatformDetection), nameof(PlatformDetection.IsSpeedOptimized), nameof(PlatformDetection.Is64BitProcess))] // don't run it on slower runtimes
+        [OuterLoop("It requires almost 12 GB of free disk space")]
+        public static void UnzipOver4GBZipFile()
+        {
+            byte[] buffer = GC.AllocateUninitializedArray<byte>(1_000_000_000); // 1 GB
```

Comment:
Technically, this is not 1Gb, but since you are creating 6 files, the test is still valid. 

I would rather create two tests, one that succeeds at the limit (say 3.999 Gb) without the fix and one that fails and then verify that both pass with your fix. 

Since this is urgent for a backport, we can defer that.

Reply:
> Technically, this is not 1Gb, but since you are creating 6 files, the test is still valid.

Reply:
> Technically, this is not 1Gb, but since you are creating 6 files, the test is still valid.

You mean that $1000^3$ is not 1 GB but $1024^3$ is?

> I would rather create two tests, one that succeeds at the limit (say 3.999 Gb) without the fix and one that fails and then verify that both pass with your fix.

My typical workflow is to add test that fails first, then make it pass. We have plenty of tests that verify < 4 GB archives so I don't see the value in adding it (also it would require us to use even more disk space and take longer to execute). 

In the long term we could add more tests, especially such that verify that `decompress(compress(x)) == x` (which this test avoids, as it was taking 10 minutes to do so on my beefy PC).

Reply:
> You mean that $1000^3$ is not 1 GB but $1024^3$ is?

Yes.

> We have plenty of tests that verify < 4 GB archives so I don't see the value in adding it (also it would require us to use even more disk space and take longer to execute).

I meant to verify on the edge cases of the bug. In this case that would be 4Gb - 1, 4Gb, and 4Gb + 1; rather than a 6Gb case. But yes, that would multiply test execution time which is already quite large with this new test.